mlflow/deployments/databricks/__init__.py

```diff
@@ -1,39 +1,123 @@
+import posixpath
+from typing import Any, Dict, Optional
+
 from mlflow.deployments import BaseDeploymentClient
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.utils.rest_utils import augmented_raise_for_status, http_request
 
 
 class DatabricksDeploymentClient(BaseDeploymentClient):
     def create_deployment(self, name, model_uri, flavor=None, config=None, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def update_deployment(self, name, model_uri=None, flavor=None, config=None, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def delete_deployment(self, name, config=None, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def list_deployments(self, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
     def get_deployment(self, name, endpoint=None):
+        """
+        .. warning::
+
+            This method is not implemented for Databricks deployments.
+        """
         raise NotImplementedError
 
+    def _call_endpoint(
+        self,
+        *,
+        method: str,
+        prefix: str = "/api/2.0",
+        route: Optional[str] = None,
+        json_body: Optional[Dict[str, Any]] = None,
+    ):
+        call_kwargs = {}
+        if method.lower() == "get":
+            call_kwargs["params"] = json_body
+        else:
+            call_kwargs["json"] = json_body
+
+        response = http_request(
+            host_creds=get_databricks_host_creds("databricks"),
+            endpoint=posixpath.join(prefix, "serving-endpoints", route if route else ""),
+            method=method,
+            timeout=10,
```

Comment:
This timeout is too low for predict - can we use the same timeout used by gateway client? It can take O(minutes) to compute completions on a large amounts of text

For CRUD, I think we should retry with exponential backoff with the same # of retries & same timeout as we use for gateway (I think it's more than a 10 second timeout)

Reply:
gateway uses 300. I'll use it.